app/vmui/packages/vmui/src/pages/ExploreAlerts/ExploreNotifiers.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { FC, useCallback, useEffect, useMemo, useState } from "preact/compat";
+import { useLocation } from "react-router";
+import { useNotifiersSetQueryParams as useSetQueryParams } from "./hooks/useSetQueryParams";
+import Spinner from "../../components/Main/Spinner/Spinner";
+import Alert from "../../components/Main/Alert/Alert";
+import Accordion from "../../components/Main/Accordion/Accordion";
+import { useFetchNotifiers } from "./hooks/useFetchNotifiers";
+import "./style.scss";
+import NotifiersHeader from "../../components/ExploreAlerts/NotifiersHeader";
+import NotifierHeader from "../../components/ExploreAlerts/NotifierHeader";
+import Notifier from "../../components/ExploreAlerts/Notifier";
+import { Notifier as APINotifier, Target as APITarget } from "../../../types";
+import { getQueryStringValue } from "../../utils/query-string";
+
+const defaultTypesStr = getQueryStringValue("types", "") as string;
+const defaultTypes = defaultTypesStr.split(",").filter(rt => rt) as string[];
+const defaultSearchInput = getQueryStringValue("q", "") as string;
+
+const ExploreNotifiers: FC = () => {
+  const { notifiers, isLoading: loadingNotifiers, error: errorNotifiers } = useFetchNotifiers();
+
+  const [expandNotifiers, setExpandNotifiers] = useState<Set<string>>([]);
+  const [searchInput, setSearchInput] = useState(defaultSearchInput);
+  const [types, setTypes] = useState(defaultTypes);
+
+  useSetQueryParams({
+    types: types.join(","),
+  });
+
+  const { hash } = useLocation();
+
+  const isLoading = useMemo(() => {
+    return loadingNotifiers;
+  }, [loadingNotifiers]);
+
+  const error = useMemo(() => {
+    return errorNotifiers;
+  }, [errorNotifiers]);
+
+  if (!isLoading && notifiers && hash && !error) {
+    useEffect(() => {
+      const target = document.querySelector(hash);
+      if (target) {
+        let parent = target.closest("details");
+        while(parent) {
+          parent.open = true;
+          parent = parent.parentElement.closest("details");
+        }
+        target.scrollIntoView();
+      }
+    }, [hash]);
+  }
```

Comment:
It's better to move condition inside of `useEffect`
```suggestion
    useEffect(() => {
      if (!isLoading && notifiers && hash && !error) {
        const target = document.querySelector(hash);
        if (target) {
          let parent = target.closest("details");
          while(parent) {
            parent.open = true;
            parent = parent.parentElement.closest("details");
          }
          target.scrollIntoView();
        }
      }
    }, [hash, isLoading, notifiers, error]);
```
Because it could throw an error , something like `Hooks can only be called at the top level...` . It is also related to other hooks too.